File: com.systemincloud.examples.tasks.pythontask/src/test/py/tasks/data/DataBool.py
from sicpythontask.PythonTaskInfo import PythonTaskInfo
from sicpythontask.PythonTask import PythonTask
from sicpythontask.InputPort import InputPort
from sicpythontask.OutputPort import OutputPort
from sicpythontask.data.Bool import Bool

@PythonTaskInfo
class DataBool(PythonTask):

    def __init_ports__(self):
        self.in1 = InputPort(name="in1", data_type=Bool)
        self.in2 = InputPort(name="in2", data_type=Bool)
        self.out = OutputPort(name="out", data_type=Bool)

    def execute(self, grp):
        in1Bool = self.in1.get_data(Bool)
        in2Bool = self.in2.get_data(Bool)
        self.out.put_data(Bool(in1Bool.values[0] and in2Bool.values[0]))

File: com.systemincloud.examples.tasks.javatask/src/test/java/tasks/data/DataBool.java
package tasks.data;

import com.systemincloud.modeler.tasks.javatask.api.InputPort;
import com.systemincloud.modeler.tasks.javatask.api.JavaTask;
import com.systemincloud.modeler.tasks.javatask.api.OutputPort;
import com.systemincloud.modeler.tasks.javatask.api.annotations.InputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.JavaTaskInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.OutputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.data.Bool;

@JavaTaskInfo
public class DataBool extends JavaTask {

	@InputPortInfo(name = "In1", dataType = Bool.class)
	public InputPort in1;
	@InputPortInfo(name = "In2", dataType = Bool.class)
	public InputPort in2;
	@OutputPortInfo(name = "Out", dataType = Bool.class)
	public OutputPort out;

	@Override
	public void execute(int grp) {
		Bool in1Bool = in1.getData(Bool.class);
		Bool in2Bool = in2.getData(Bool.class);
		out.putData(new Bool(in1Bool.getValues()[0] && in2Bool.getValues()[0]));
	}
}
